docs/antrea-network-policy.md

```diff
@@ -1433,6 +1474,101 @@ The following kubectl commands can be used to retrieve CG resources:
     kubectl get cg.crd.antrea.io
 ```
 
+## Group
+
+A Group (GRP) CRD is a another specification of how workloads are grouped together,
+which is similar to the ClusterGroup CRD. The purpose of introducing this CRD is for
+Antrea NetworkPolicy users to also leverage the advantage of grouped resources.
+
+### Group CRD
+
+Below are some example Group specs:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-sel
+spec:
+  podSelector:
+    matchLabels:
+      role: db
+  namespaceSelector:
+    matchLabels:
+      env: prod
+---
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-with-namespace
+spec:
+  podSelector:
+    matchLabels:
+      role: db
+  namespaceSelector:
+    matchLabels:
+      env: prod
+---
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-ip-block
+spec:
+  # IPBlocks cannot be set along with PodSelector, NamespaceSelector or serviceReference.
+  ipBlocks:
+    - cidr: 10.0.10.0/24
+---
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-svc-ref
+spec:
+  # ServiceReference cannot be set along with PodSelector, NamespaceSelector or ipBlocks.
+  serviceReference:
+    name: test-service
+    namespace: default
+---
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-nested
+spec:
+  childGroups: [test-grp-sel, test-grp-ip-blocks, test-grp-svc-ref]
+```
+
+### Restrictions and Key differences from ClusterGroup
+
+Group has a similar spec with ClusterGroup. However, however there exist key different
+restrictions.
+
+- A Group can be set in an Antrea NetworkPolicy's `appliedTo` and `to`/`from` peers.
+  When set in ANP `appliedTo` field cannot reference to Groups with
+  `namespaceSelector` set, since ANP is namespace scoped. For example,
+  the `test-grp-with-namespace` Group in the [sample](#group-crd) cannot be referenced
+  by ANP `appliedTo`.
+- Antrea will not enforce this validation, but only provides an API status update.
+  If an Antrea NetworkPolicy has `appliedTo` field referencing Groups with
+  `namespaceSelector` set, or all rules violates the `appliedTo` convention, Antrea
+  will provide an API status update as `Unrealizable`. If an Antrea NetworkPolicy
+  includes some valid rules but the remaining rules violate this `appliedTo`
+  convention (refer to Groups with `namespaceSelector`set), Antrea will provide an API
+  status update as `PartiallyRealizable`.
```

Comment:
I had to read this paragraph several times to understand its meaning. Perhaps we can consider simplifying a little as follows:

Antrea will not validate referenced GRP resources; it will however report an error in the ANP status if the GRP does not meet the AppliedTo Convention
  - If the convention is violated in the ANP's "AppliedTo" section or for all the rules in the ANP, then the ANP status will be updated to "Unrealizable"
  -  If the convention is violated for part of, but not all, ANP rules, the ANP status will be updated to "Partially Realizable""

Reply:
I did not see Salvatore's comment when adding my comments above. Salvatore's version looks good to me.